File: 0438-find-all-anagrams-in-a-string/0438-find-all-anagrams-in-a-string.java
class Solution {
    public List<Integer> findAnagrams(String s, String p) {
      int[] count=new int[26];
        List<Integer> ans=new ArrayList();
        for(char c:p.toCharArray()){
            count[c-'a']++;
        }
        
        int l=0;
        int n=p.length();
        int length=0;
        int r=0;
       while(r<s.length()){
           if(count[s.charAt(r++)-'a']-->=1)n--;
           if(n==0) ans.add(l);
           if(r-l==p.length()&&count[s.charAt(l++)-'a']++>=0){
               System.out.println(r);
               n++;
           }
           
       }
        return ans;
    }
}

File: 0438-find-all-anagrams-in-a-string/0438-find-all-anagrams-in-a-string.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        k=len(p)
        pov=sorted(list(p))
        s=list(s)
        ans=[]
        l=0
        r=k
        while r<=len(s):
            sub=sorted(s[l:r])
            if pov==sub:
                ans.append(l)
            r+=1
            l+=1
        return ans
            